### PR TITLE
Update to Weerlive API v2

### PR DESCRIPTION
**Pull request by golles**
If you are willing to test out this update, I would really appreciate your feedback!

See the draft release notes in my comment below, and the updated [README.md](https://github.com/golles/ha-knmi/blob/api-v2/README.md) as there are some breaking changes.
Please answer the following questions:

1. What do you think about entity and attribute naming? 
see [en.json](https://github.com/golles/ha-knmi/blob/api-v2/custom_components/knmi/translations/en.json#L30-L94) and [nl.json](https://github.com/golles/ha-knmi/blob/api-v2/custom_components/knmi/translations/nl.json#L30-L94), please comment in the pull request files section to keep it organized
2. Are there sensors that are disabled by default that should be enabled by default?
3. How painful is the breaking change where entity names and id's are changed?
4. Anything else on your mind?

Tasks list:

- [x] Weather
  - [x] Implement 5-day daily forecast
  - [x] Implement 24-hour hourly forecast
- [x] Binary sensors and sensors
  - [x] Add more sensors
  - [x] Improve SensorDescription with `value_fn` and `attr_fn` instead of subclassing = less code
  - [x] A better way to get values from the API and handling of time values
- [ ] Work out a migration path (Note: Done but not perfect as old entities are removed...)
- [x] Update readme 
  - [x] General content
  - [x] Explain timestamp
  - [x] Explain how to enable sensors
  - [x] Screenshots
- [x] Testing
  - [x] More test cases, each sensor state/attributes
  - [x] Write a detailed test for `async_migrate_entry`
  - [x] Run in my production HA
  - [x] Ask for feedback

Known bugs:
- [x] Warning sensor, next warning attribute isn't working properly yet date (`wrsch_g`) and timestamp (`wrsch_gts`) format from the API are inconsistent, if `wrsch_gts` becomes an `int`, no action is requited.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,48 +15,114 @@ Weather data provided by KNMI, https://weerlive.nl.
...
+| Name (EN)              | Name (NL)                | Notes                   |
+| ---------------------- | ------------------------ | ----------------------- |
+| Location               | Plaats                   |                         |
+| Remaining API requests | Resterende API verzoeken |                         |
+| Latest update          | Laatste update           | Server side update time |
+
+### Weather
+
+The weather entity holds all the necessary weather information, this can be nicely shown in the Home Assistant frontend.
+There are is a daily forecast, up to 5 days, and an hourly forecast, up to 24 hour.
```
Comment (golles) [inline README.md:95]:
```suggestion
The weather entity contains all the weather information, ideal for displaying a comprehensive overview in the Home Assistant frontend. It includes both a daily forecast spanning up to 5 days and an hourly forecast covering up to 24 hours.
```

**Comment by golles** [comment]
Scherm­afbeelding 2024-02-17 om 19 08 04

Old version number persisted in device info

**Comment by golles** [comment]
Scherm­afbeelding 2024-02-17 om 19 08 11

API is returning old data, I could consider a workaround for daily and hourly forecasts that items in the past shouldn't be added. Will inform supplier as well

**Comment by golles** [comment]
I see that HA also removes entities on platform updates, like recently https://github.com/home-assistant/core/pull/110371/commits/54152433ef06e77682c8910db166e44beeeb8285
This code is better than in this PR, so should do it like that.

File: custom_components/knmi/__init__.py [hunk 2]
```diff
@@ -55,9 +56,9 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     device_info = DeviceInfo(
         entry_type=DeviceEntryType.SERVICE,
         identifiers={(DOMAIN, entry.entry_id)},
-        manufacturer=NAME,
+        manufacturer=SUPPLIER,
```
Comment (golles) [inline custom_components/knmi/__init__.py:59]:
```suggestion
        model=NAME,
        manufacturer=SUPPLIER,
```

File: custom_components/knmi/sensor.py [hunk 3]
```diff
@@ -18,43 +30,202 @@
...
         device_class=SensorDeviceClass.TEMPERATURE,
         state_class=SensorStateClass.MEASUREMENT,
+        translation_key="gtemp",
+        value_fn=lambda coordinator: coordinator.get_value(["liveweer", 0, "gtemp"]),
+        entity_registry_enabled_default=False,
     ),
-    SensorEntityDescription(
+    KnmiSensorDescription(
+        key="luchtd",
+        icon="mdi:water-percent",
```
Comment (golles) [inline custom_components/knmi/sensor.py:73]:
```suggestion
```

**Comment by martijnrusschen** [comment]
This is really interesting thanks for taking the effort to migrate the APIs. Would it be possible to make a pre-release out of this? That makes it easier to test the changes using HACS.

**Comment by golles** [comment]
> This is really interesting thanks for taking the effort to migrate the APIs. Would it be possible to make a pre-release out of this? That makes it easier to test the changes using HACS.

Thanks, yes will make a pre-release later today

**Comment by hmmbob** [comment]
Works fine so far, and thanks for the seperate sensors - it simplifies my code big time.

Q: did `sensor.knmi_korte_dagverwachting` get dropped? I used it in my morning briefing :)
Of course there is `sensor.knmi_omschrijving`, but that tends to me more verbose.

**Comment by martijnrusschen** [comment]
Working as expected here for the past few days. I like the additional sensors and forecasts. 

q: Is there a way to get wind direction forecast? I was able to extract a separate sensor for the current wind direction, but would be great to get more wind direction details if possible.

**Comment by golles** [comment]
> Works fine so far, and thanks for the seperate sensors - it simplifies my code big time.
> 
> Q: did `sensor.knmi_korte_dagverwachting` get dropped? I used it in my morning briefing :) Of course there is `sensor.knmi_omschrijving`, but that tends to be more verbose.

This value is now in the sensor with the name Weather forecast/Weersverwachting

**Comment by golles** [comment]
> q: Is there a way to get wind direction forecast? I was able to extract a separate sensor for the current wind direction, but would be great to get more wind direction details if possible.

I think this is too specific, for this, I would suggest to use the `weather.get_forecasts` service with either hourly or daily as a type

**Pull request merged**